subiquitycore/ui/form.py

```diff
@@ -504,13 +517,13 @@ def as_screen(self, focus_buttons=True, excerpt=None, narrow_rows=False):
             focus_buttons=focus_buttons, excerpt=excerpt,
             narrow_rows=narrow_rows)
 
+    def has_validation_error(self, recurse: bool = True):
+        """ Tells if any field (or subformfield if recurse is True) is in
+        error. """
+        return any(map(lambda f: f.is_in_error(recurse=recurse), self._fields))
+
     def validated(self):
-        in_error = False
-        for f in self._fields:
-            if f.in_error:
-                in_error = True
-                break
-        if in_error:
+        if self.has_validation_error(recurse=True):
```

Comment:
Just curious - can you tell me about the use case of recurse=False?

Reply:
Yeah looks like this argument is no longer needed?

Reply:
Yes, we can get rid of it now. It's pretty useful in the debugger to figure out why a form would enable/disable the Done button. But yeah, not needed in the codebase. I'll remove it.